drivers/timer/cortex_m_systick.c

```diff
@@ -202,7 +203,13 @@ void z_clock_set_timeout(s32_t ticks, bool idle)
 		delay =
 		 ((delay + CYC_PER_TICK - 1) / CYC_PER_TICK) * CYC_PER_TICK;
 		delay -= unannounced;
-		last_load = delay;
+		if (delay <= MAX_CYCLES) {
+			last_load = delay;
+		} else {
+			/* If delay contains a value that cannot be programmed,
+			 * leave the previous value of last_load into SysTic.
```

Comment:
Why leave this at whatever value it was before, rather than do:

    last_load = MAX_CYCLES;

to sleep for the longest allowable duration?  Also this is now inconsistent with the commit message which describes that behavior.

Reply:
I've been experimenting on this.
i think , if we reach a point that we provide trash to the systick LOAD register, I am not sure what the best policy should be.

Reply:
But it's not "trash", it's simply a deadline that's too large to fit in the limited number of bits in `LOAD` and so has reached in steps.  The maximum tick-aligned value that does fit seems like the best policy; I don't know of another that would be defensible.  (Consider a desire to sleep for 2^25 cycles, but the last completed timeout was for 2^8 cycles.  The systick interrupt would have to fire 2^16 times to reduce the sleep to 2^24 which could be done in one step.)